modules/images/webp-uploads/load.php

```diff
@@ -529,17 +529,18 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 
 	$basename = wp_basename( $metadata['file'] );
 	foreach ( $urls as $url ) {
-		if ( isset( $metadata['file'] ) && strpos( $url, $basename ) !== false ) {
-			// TODO: we don't have a replacement for full image yet, issue. See: https://github.com/WordPress/performance/issues/174.
-			continue;
+		$src_filename = wp_basename( $url );
+
+		// Replace the full size image if present.
+		if ( isset( $metadata['file'], $metadata['sources'][ $target_mime ]['file'] ) && strpos( $url, $basename ) !== false ) {
+			$image = str_replace( $src_filename, $metadata['sources'][ $target_mime ]['file'], $image );
```

Comment:
I think this is missing a `continue` statement? If we replaced the full image, we already took care of this URL, so we should skip the rest of this code block to proceed with the next URL.
```suggestion
			$image = str_replace( $src_filename, $metadata['sources'][ $target_mime ]['file'], $image );
			continue;
```

Also, please review the related line 565 below (I can't comment there since it wasn't changed here): I believe we should put a `break;` right after the image replacement, otherwise we'll always run through all sizes even if we already made a successful replacement, which would be pointless. This is also something which we should have caught before, but better now than never :)

Reply:
Good catch :+1:

Reply:
Note the second part of my comment as well. Your change https://github.com/WordPress/performance/pull/195/commits/7e1b5298b22f784feabde43b66e8191bafa5d6a1 adds the `continue` here, but not the `break` further down in the code.

Reply:
Right, let me update that one.